### ISSUE TITLE
Metrics question

### ISSUE THREAD

**Issue by ShawnyXiao** [keras-team/keras#11512]
It's a question about metrics execution time. When I execute code like:

```
model.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])
model.fit(x_train, y_train, validation_data=(x_test, y_test))
```

I wanna know what time the metrics function would be executed on **validation set**, end of a batch or end of a epoch?

Thanks!

**Comment by steeveen** [comment]
epoch

**Comment by ShawnyXiao** [comment]
I try to do a experiment on this question. I implement a epoch-level f1 metric inherited from Callbacks, which is compared by specifying customed f1 metric at compiling model. The process of training is as follow:

![image](https://user-images.githubusercontent.com/9656978/47719872-6c45a780-dc87-11e8-8583-42e323b46938.png)

They are very inconsistent. The red square frame is f1 metric inherited from Callbacks. The blue square frame is specifying customed f1 metric at compiling model. 

From the picture, it seem like that:
1. When specifying metrics at compiling model, the metrics function would be executed at end of a batch. 
2. When implement metrics inherited from Callbacks, you could custom the execution time of metrics function, either at end of a batch or at end of a epoch.
